File: tests/fixtures/anti-patterns/typescript/console-log-vulnerable.ts
/**
 * Test Fixture: Console Log Vulnerable Patterns (TypeScript)
 *
 * This file contains console.log statements in production code.
 * Expected detections: >=3 violations for AP-003
 * Rule ID: AP-003
 * Severity: MEDIUM (info)
 */

interface User {
  id: number;
  name: string;
  email: string;
}

function processUserData(user: User): User {
  console.log("Processing user:", user); // VULNERABLE: Debug log

  if (!user.email) {
    console.warn("User has no email"); // VULNERABLE: Should use proper logging
  }

  const result = transformUser(user);
  console.log(`Transformed result: ${JSON.stringify(result)}`); // VULNERABLE: Debug log

  return result;
}

function transformUser(user: User): User {
  console.debug("Transforming user"); // VULNERABLE: Debug log
  return { ...user, name: user.name.toUpperCase() };
}

function handleError(error: Error): void {
  console.error("An error occurred:", error); // VULNERABLE: Should use logging service
}

export { processUserData, transformUser, handleError };

File: tests/fixtures/anti-patterns/python/console_log_vulnerable.py
"""
Test Fixture: Console Log / Print Vulnerable Patterns (Python)

This file contains print() statements that should be caught in production code.
Expected detections: >=3 violations for AP-003
Rule ID: AP-003
Severity: MEDIUM (info)
"""


def process_user_data(user):
    """Process user data with debug prints."""
    print("Processing user:", user)  # VULNERABLE: Debug print

    if not user.get("email"):
        print("Warning: User has no email")  # VULNERABLE: Should use logging

    result = transform_user(user)
    print(f"Transformed result: {result}")  # VULNERABLE: Debug print

    return result


def transform_user(user):
    """Transform user data."""
    return {"id": user.get("id"), "name": user.get("name")}
